x-pack/solutions/security/plugins/elastic_assistant/server/routes/knowledge_base/get_knowledge_base_indices.ts

```diff
@@ -44,23 +46,71 @@ export const getKnowledgeBaseIndicesRoute = (router: ElasticAssistantPluginRoute
         const esClient = ctx.core.elasticsearch.client.asCurrentUser;
 
         try {
-          const body: GetKnowledgeBaseIndicesResponse = {
-            indices: [],
-          };
-
           const res = await esClient.fieldCaps({
             index: '*',
             fields: '*',
-            types: ['semantic_text'],
+            types: ['sparse_vector'],
             include_unmapped: true,
+            filter_path: 'fields.*.sparse_vector.indices',
+          });
+
+          const indicesWithSemanticTextField = reduce(
+            res.fields,
+            (acc, value, key) => {
+              if (key.endsWith('.inference.chunks.embeddings')) {
+                forEach(value?.sparse_vector?.indices ?? [], (index) => acc.add(index));
+              }
+              return acc;
+            },
+            new Set<string>()
+          );
+
+          const mappings = await esClient.indices.getMapping({
+            index: Array.from(indicesWithSemanticTextField),
+            filter_path: '*.mappings.properties',
           });
 
-          body.indices = Object.values(res.fields)
-            .flatMap((value) => value.semantic_text?.indices ?? [])
-            .filter((value, index, self) => self.indexOf(value) === index)
-            .sort();
+          const findSemanticTextPaths = (
+            obj: IndicesGetMappingIndexMappingRecord,
+            currentPath: string[] = []
+          ): string[][] => {
+            let paths: string[][] = [];
+
+            for (const key in obj) {
+              if (Object.prototype.hasOwnProperty.call(obj, key)) {
+                // @ts-expect-error
+                const value = obj[key];
+                const newPath = currentPath.concat(key);
+
+                if (value && typeof value === 'object') {
+                  if (value.type === 'semantic_text') {
+                    paths.push(
+                      newPath.filter(
+                        (pathPart) => pathPart !== 'properties' && pathPart !== 'mappings'
```

Comment:
are `properties` and `mappings` both reserved names and never can be field names?